pkg/auth/handler/forgotpwd/forgot_password.go

```diff
@@ -147,7 +151,7 @@ func (h ForgotPasswordHandler) Handle(req interface{}) (resp interface{}, err er
 		if err = h.ForgotPasswordEmailSender.Send(
 			payload.Email,
 			fetchedAuthInfo,
-			userProfile,
+			userProfile.MergeLoginIDs(loginIDs),
```

Comment:
Merging login ID to user profile here looks inconsistent to me, because this only appears in email template. Developers do not see this in the sdk. Also if we want to promote meaningless login IDs, this should not be mixed with user profile.

I just revised the forgot password flow. There are two places where the user profile is used.

- The code generation does not need `userprofile` at all, it just need email.
- The `user` in email template context was `userprofile`, the equivalent now should be user metadata, however, if we do not encourage developers to put application data in the metadata, I think putting the whole user object would be more useful.